app/sources/HFTclTemplateController.m

```diff
@@ -612,16 +612,21 @@ - (int)runTypeCommand:(enum command)command objc:(int)objc objv:(struct Tcl_Obj
             const char *arg = Tcl_GetStringFromObj(extraArgs[i], NULL);
             if (arg && arg[0] == '-') {
                 Tcl_SetObjResult(_interp, Tcl_ObjPrintf("Unknown option %s", arg));
+                ckfree((char *)extraArgs);
```

Comment:
It seems to me this will ckfree the pointer to the extra args array for each of its members you loop through.

Reply:
We're doing `return TCL_ERROR` just below. Should be ok then?